src/Network/Network/VirtualRouter/GetAzureRMVirtualRouterCommand.cs

```diff
@@ -0,0 +1,118 @@
+        public string ResourceId { get; set; }
+
+        public override void Execute()
+        {
+            base.Execute();
+
+            if (string.Equals(this.ParameterSetName, VirtualRouterParameterSetNames.ByVirtualRouterResourceId, StringComparison.OrdinalIgnoreCase))
+            {
+                var resourceInfo = new ResourceIdentifier(ResourceId);
+                ResourceGroupName = resourceInfo.ResourceGroupName;
+                RouterName = resourceInfo.ResourceName;
+            }
+
+            if (ShouldGetByName(ResourceGroupName, RouterName))
+            {
+                var vVirtualRouter = this.NetworkClient.NetworkManagementClient.VirtualRouters.Get(ResourceGroupName, RouterName);
+                var vVirtualRouterModel = NetworkResourceManagerProfile.Mapper.Map<CNM.PSVirtualRouter>(vVirtualRouter);
+                vVirtualRouterModel.ResourceGroupName = this.ResourceGroupName;
+                vVirtualRouterModel.Tag = TagsConversionHelper.CreateTagHashtable(vVirtualRouter.Tags);
+                foreach (var peering in vVirtualRouter.Peerings)
+                {
+                    var resourceInfo = new ResourceIdentifier(peering.Id);
+                    var vVirtualRouterPeering = this.NetworkClient.NetworkManagementClient.VirtualRouterPeerings.Get(ResourceGroupName, RouterName, resourceInfo.ResourceName);
+                    vVirtualRouterModel.Peerings.Add(NetworkResourceManagerProfile.Mapper.Map<CNM.PSVirtualRouterPeer>(vVirtualRouterPeering));
+                }
+                WriteObject(vVirtualRouterModel, true);
+            }
+            else
+            {
+                IPage<VirtualRouter> vVirtualRouterPage;
+                if(ShouldListByResourceGroup(ResourceGroupName, RouterName))
+                {
+                    vVirtualRouterPage = this.NetworkClient.NetworkManagementClient.VirtualRouters.ListByResourceGroup(this.ResourceGroupName);
+                }
+                else
+                {
+                    vVirtualRouterPage = this.NetworkClient.NetworkManagementClient.VirtualRouters.List();
+                }
+
+                var vVirtualRouterList = ListNextLink<VirtualRouter>.GetAllResourcesByPollingNextLink(vVirtualRouterPage,
+                    this.NetworkClient.NetworkManagementClient.VirtualRouters.ListNext);
+                List<PSVirtualRouter> psVirtualRouterList = new List<PSVirtualRouter>();
+                foreach (var vVirtualRouter in vVirtualRouterList)
+                {
+                    RouterName = vVirtualRouter.Name;
+                    var vVirtualRouterModel = NetworkResourceManagerProfile.Mapper.Map<CNM.PSVirtualRouter>(vVirtualRouter);
+                    vVirtualRouterModel.ResourceGroupName = NetworkBaseCmdlet.GetResourceGroup(vVirtualRouter.Id);
+                    vVirtualRouterModel.Tag = TagsConversionHelper.CreateTagHashtable(vVirtualRouter.Tags);
+                    foreach (var peering in vVirtualRouter.Peerings)
```

Comment:
Same comment about using `VirtualRouterPeerings.List`

Reply:
Agreed. Good idea to reduce network calls.Done